### PR TITLE
Update xml_response.rb

### PR DESCRIPTION
**Pull request by picman**
Ruby 3.0 compatibility

### CODE REVIEW

File: lib/dav4rack/xml_response.rb [hunk 1]
```diff
@@ -71,35 +68,28 @@ def render_lockdiscovery(*args)
 
 
     # returns an activelock Ox::Element for the given lock data
-    def activelock(time: nil, token:, depth:,
-                      scope: nil, type: nil, owner: nil, root: nil)
-
+    def activelock(options)
```
Comment (jkraemer) [inline lib/dav4rack/xml_response.rb]: I would really prefer to not go back from keyword arguments to an options hash.

File: lib/dav4rack/xml_response.rb [hunk 2]
```diff
@@ -57,11 +57,8 @@ def render_lock_errors(errors)
     end
 
 
-    def render_lockdiscovery(*args)
```
Comment (jkraemer) [inline lib/dav4rack/xml_response.rb:60]: how about `def render_lockdiscovery(**options)` ?

File: lib/dav4rack/xml_response.rb (hunk 1 above)
Comment (picman) [reply lib/dav4rack/xml_response.rb]: Okay, reverted.

File: lib/dav4rack/xml_response.rb (hunk 2 above)
Comment (picman) [reply lib/dav4rack/xml_response.rb:60]: The double splat operator used.

**Comment by picman** [comment]
- List of parameters kept.
- Double splat operator used.
- Successfully tested with ruby 3.0.0 and 2.6.3

File: lib/dav4rack/xml_response.rb [hunk 3]
```diff
@@ -71,35 +68,28 @@ def render_lockdiscovery(*args)
...
 
     # returns an activelock Ox::Element for the given lock data
-    def activelock(time: nil, token:, depth:,
-                      scope: nil, type: nil, owner: nil, root: nil)
-
+    def activelock(time: nil, token:, depth:, scope: nil, type: nil, owner: nil, root: nil)
       Ox::Element.new(D_ACTIVELOCK).tap do |activelock|
         if scope
-          activelock << ox_element(D_LOCKSCOPE, scope)
+          value = Ox::Element.new("#{DAV_NAMESPACE_NAME}:#{scope}")
```
Comment (jkraemer) [inline lib/dav4rack/xml_response.rb:74]: It is a good idea to do this in here, but it's a breaking change (I am looking at code that calls this with the namespace prepended already right now). Also has nothing to do with Ruby 3 compatibility and should probably be in a separate PR that can be referenced from the changelog.

I'd opt for adding a check if the namespace is present already.

File: lib/dav4rack/xml_response.rb [hunk 4]
```diff
@@ -71,35 +68,28 @@ def render_lockdiscovery(*args)
...
+          activelock << ox_element(D_LOCKSCOPE,  value)
         end
         if type
-          activelock << ox_element(D_LOCKTYPE, type)
+          value = Ox::Element.new("#{DAV_NAMESPACE_NAME}:#{type}")
```
Comment (jkraemer) [inline lib/dav4rack/xml_response.rb:78]: same as with `scope`

**Review by jkraemer — COMMENTED** [review]
Sorry for being so picky ;)

While looking at the code now (it's been a while...) I also noticed that there is quite some duplication between `xml_elements.rb` and `xml_response.rb`. The original plan was to get rid of the module with helpers and do all XML rendering in the xml response class. But this never got anywhere so currently there is `ox_activelock` which is used in `resource.rb` for building a `PROPFIND` response, and `activelock` which is used when rendering the `LOCK` response.

I think for the time being and since `XmlElements` is included in `XmlResponse` anyway it would be good to move your updated `activelock` method back into the xml_elements module (as `ox_activelock`, replacing the old version) and use it from there. I would do this after merging your PR, or you can do it while you're at it...

**Comment by picman** [comment]
There is a separate pull request, from February....
https://github.com/planio-gmbh/dav4rack/pull/6

**Comment by picman** [comment]
I'm not so familiar with your code. Please merge my MRs and finalize the code according to your suggestions. If I do that you will have for sure some objections again. My intentions is to use this library in DMSF and not to keep separate code with my own changes.

**Comment by jkraemer** [comment]
tests now pass with Ruby 3

**Comment by picman** [comment]
Great work. Thank you.
